src/field/field_factory.cxx

```diff
@@ -231,6 +232,8 @@ Field3D FieldFactory::create3D(FieldGeneratorPtr gen, Mesh* localmesh, CELL_LOC
       // Transform from field aligned coordinates, to be compatible with
       // older BOUT++ inputs. This is not a particularly "nice" solution.
       result = fromFieldAligned(result, "RGN_ALL");
+    } else {
+      result.setDirectionY(YDirectionType::Standard);
```

Comment:
I guess the purpose of this is to handle the default of `transform_from_field_aligned` being `true`, while this is not possible for some `ParallelTransform`s. But if the user has explicitly asked for `transform_from_field_aligned = true` with `FCITransform`, I'd expect an error because it's not possible. The nicest solution would be to default to `transform_from_field_aligned = false` but of course that's not backward-compatible; would the next-best thing be to have the default be `transform_from_field_aligned = coords->getParallelTransform().canToFromFieldAligned()`? Then in this `else` we could throw an exception because we'd only reach it if the user has asked for something impossible.

Reply:
This turns out to be much more difficult that I first thought. The default for `transform_from_field_aligned` should be set in the constructor for `FieldFactory`. In order to call `canToFromFieldAligned()`, we need the `Coordinates`, but to make that we need the regions to be set up already.  
For `BoutMesh`, this is done at the _end_ of `BoutMesh::load`. Unfortunately, we may need `FieldFactory` at the _beginning_ in order to call `Mesh::get(nx, "nx")`.

So we've got a circular dependency if we want to do this.

I think I'd prefer to leave it for the time being and hopefully come with something clever to fix it in the future.